firestore-translate-text/functions/src/index.ts

```diff
@@ -144,11 +224,25 @@ const handleUpdateDocument = async (
 
 const translateSingle = async (
   input: string,
-  snapshot: admin.firestore.DocumentSnapshot
+  snapshot: admin.firestore.DocumentSnapshot,
+  keepExistingTranslations: boolean = false
 ): Promise<void> => {
-  logs.translateInputStringToAllLanguages(input, config.languages);
+  const existingTranslations = extractOutput(snapshot);
+  const languages = config.languages.filter(
+    (targetLanguage: string): boolean => {
+      if (
+        !keepExistingTranslations ||
+        existingTranslations[targetLanguage] != undefined
```

Comment:
This if condition surprises me. Maybe I'm missing something, but I would have expected:

If {keepExistingTranslations && existingTranslations[targetLanguage] != undefined} then skip language.

Reply:
Good catch - i got this mixed up while i was adding logging.